plumpy/processes.py

```diff
@@ -1193,6 +1193,12 @@ async def step(self) -> None:
 
             except KeyboardInterrupt:  # pylint: disable=try-except-raise
                 raise
+            except asyncio.CancelledError:  # pylint: disable=try-except-raise
```

Comment:
Just wondering: is cancellation conceptually also an interruption or should these two be treated differently?

In the other two cases above you're capturing the `CancelledError` together with the `Interruption`.
That makes me wonder whether `Interruption` should inherit from `asyncio.CancelledError` and we just generalize `_set_interrupt_action_from_exception` to deal properly also with that case?

Perhaps a question for @muhrin 

Don't know whether this makes sense, I'm just not clear on what 

also an int

Reply:
Indeed, it's a bit subtle and it's difficult to give a good answer without looking through in detail all of the code you've been discussing.

However, my intuition is that `CancelledError` should _not_ be treated as an interruption.  My thinking is that, for futures (both concurrent and async), there are three terminal states (success, exception and cancelled) and ideally I would be tempted to keep them as such.  Interrupted was created precisely to give another form of intervention for our specific use and, if possible, all interruptions would be channelled through this mechanism.